conclave-host/src/main/kotlin/com/r3/conclave/host/internal/SocketHostEnclaveInterface.kt

```diff
@@ -88,7 +90,8 @@ class SocketHostEnclaveInterface(private val port: Int) : HostEnclaveInterface()
             /** Start the message receive thread and allow calls to enter. */
             receiveLoopThread.start()
             callGuardSemaphore.release(maxConcurrentCalls)
-            state = State.RUNNING
+
+            stateManager.transitionStateFrom<State.Ready>(to = State.Running)
```

Comment:
You can move this up and replace it with the `checkStateIs`.

Reply:
Is this wise? The operations in between can fail. Let me know if I'm over-thinking it.

Reply:
if `start()` fails then you don't want it to be possible to call it again.